src/object/schedcontrol.c

```diff
@@ -72,8 +72,10 @@ static exception_t invokeSchedControl_Configure(sched_context_t *target, word_t
     }
 #endif /* ENABLE_SMP_SUPPORT */
```

Comment:
"When reconfiguring an SC, refill_new and refill_update will always
produce a SC that is ready and so will never need to be postponed."

Does this mean that you can violate the bandwidth constraints on the SC by reconfiguring?

This was something the initial design tried to avoid, as there are valid cases where you want to adjust the scheduling parameters but still need to be able to to scheduling analysis.

Reply:
Ah, this comment is incorrect. `refill_update` preserves readiness, so it can't make a ready SC become not-ready.

This means the assert is invalid and the SC may need to be returned to the release queue. I don't think `refill_update` should fault though. An alternative may be to not remove it from the release queue ore ready queues at all given that `refill_update` doesn't affect the time of the head refill.

 * [ ] Fix `refill_update` changes

Reply:
I replaced the commit with this one: https://github.com/seL4/seL4/pull/216/commits/133da34c4c9fdf7468912b3391f3046361f07b4e

This keeps the effect of the `schedContext_resume` but manually inlines it and removes any unnecessary checks (although any checks removed now get asserted).